IO/Browser/Makefile.PL

```diff
@@ -20,29 +20,75 @@ $hash{'clean'}{FILES} .= ' browse$(OBJ_EXT) browse$(EXE_EXT) Browser.c Browser.p
 # (4) confirm configuration
 # (5) write Makefile or dummy as appropriate
 
-my $incstring;
-foreach my $incl ( qw( curses.h ncurses/curses.h ncurses.h ncurses/ncurses.h ncursesw/ncurses.h ) ) {
+my @possible_headers = qw( curses.h ncurses/curses.h ncurses.h ncurses/ncurses.h ncursesw/ncurses.h );
+my $found_header;
+foreach my $incl (@possible_headers) {
     if (eval { check_lib(header=>$incl) }) {
-       $incstring = $incl;
+       $found_header = $incl;
        last;
     }
-};
-$hash{DEFINE} .= ' -DCURSES=' . '\\"' . $incstring . '\\"' if defined $incstring;
-
-my $libstring;
-foreach my $libr ( qw( curses ncurses ncursesw ) ) {
-   if (eval { check_lib(lib=>$libr) }) {
-      $libstring = '-l' . $libr;
-      last;
+}
+
+sub conf_via_ncurses_config {
+   return () unless $found_header;
+
+   my @nc_config = qw(
+      ncurses6-config
+      ncursesw6-config
+      ncurses5-config
+      ncursesw5-config
+   );
+   for my $nc_conf (@nc_config) {
+      local $ENV{NCURSES_CONFIG_PROG} = $nc_conf;
```

Comment:
Using an environment variable is absolutely necessary otherwise MSYS2 will think it can rewrite the path for you which causes it to hang (I think this is why).

Reply:
Is there any way to use a list form? That `sh -c '...'` hurts my eyes ;-) Also, would it make sense to at least use File::Which first?

Reply:
I'd like `File::Which` if we can make that a configure dep (I think it is just runtime now). The only exception is that `File::Which` will not work on Windows because it only considers non-empty `PATHEXT` as executable (i.e., it has to have a suffix).

I could use the list form with `open( my $fh, "-|", ...)` even if that's a bit longer (but I'm happy to make that change). My preference is to use `Capture::Tiny`, but I didn't want to add another configure dep in this PR (even though both `File::Which` and `Capture::Tiny` are small).

Reply:
I would be Ok with F:W as a configure dep. I'm with you on preferring not to add deps. And my preference is to use the `open` list construct you give, maybe with a tiny utility function if using several times? Is it possible that would bypass the `ENV` thing as well?